### PR TITLE
[sensor-tester] Add rest of tests

### PR DESCRIPTION
**Pull request by Honry**
- Including OrientationSensor, Ambient Light Sensor, Magnetometer,
  Gyroscope
  - Use "max" and "min" keys to check range of illuminance

### CODE REVIEW

**Comment by Honry** [comment]
@alexshalamov, PTAL. 
Not sure expecting -10.0/10.0 of Magnetometer is appropriate.

File: sensor-tester/src/tests/magnetometer.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
+[
+    {
+      "name":         "Test +X magnetic north",
+      "description":  "Test that sensor reports positive magnetic field along the X axis when device is at rest on a leveled surface and its +X axis is pointing magnetic north.",
```
Comment (alexshalamov) [inline sensor-tester/src/tests/magnetometer.json]: For magnetometer, I would ask user to rotate device 360 degrees around each axis before the test, then, check x and y measurements, whenever x or y axis points toward the north, it should report around 0 microtesla. So, let's just add 2 tests, one that asks to point Y toward the North, another for X axis.

For checking if heading can be calculated correctly, I would need to add bit more code.

File: sensor-tester/src/tests/relativeorientationsensor.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+[
```
Comment (alexshalamov) [inline sensor-tester/src/tests/relativeorientationsensor.json]: This sensor does not use magnetometer, no need to point it to the north. We need to think how to test it properly, e.g., save point of reference and then ask user to rotate device, calculate difference and compare with expectation.

Let's drop this test for now, I have to think how to add support for it.

File: sensor-tester/src/tests/ambientlightsensor.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
+[
+    {
+      "name":         "Test day mode",
+      "description":  "Test that sensor reports positive illuminance when device is around a bright environment.",
+      "illustration": "src/tests/als_day.png",
+      "duration":     3,
+      "expected":     { "illuminance": "positive" },
```
[the remark this answers is not in the document]
Comment (Honry) [reply sensor-tester/src/tests/ambientlightsensor.json]: For night mode, I did some testing, 0 lux is easier to reach than 5 lux. WDYT?

File: sensor-tester/src/tests/gyroscope.json [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+      "duration":     3,
+      "expected":     { "x": 0, "y": 0, "z": 0 },
+      "epsilon":      1
+    },
+    {
+      "name":         "Test +Z angular velocity",
+      "description":  "Test that sensor reports positive angular velocity around the Z axis when rotating device counter-clockwise around its Z axis.",
+      "illustration": "src/tests/gyro_+z.gif",
+      "duration":     5,
+      "expected":     { "z": 3.0 },
```
[the remark this answers is not in the document]
Comment (Honry) [reply sensor-tester/src/tests/gyroscope.json]: Let's use 5 rad/s. :)

File: sensor-tester/src/tests/ambientlightsensor.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (Honry) [reply sensor-tester/src/tests/ambientlightsensor.json]: I am going to use `{illuminance: {"max": 5}}` and `{illuminance: {"min": 100}}` to check night mode and day mode.  WDYT?

**Comment by Honry** [comment]
@alexshalamov, thanks for review, I've addressed your comments, PTAL.

File: sensor-tester/build/bundled/src/tests/magnetometer.json [hunk 5]
```diff
@@ -0,0 +1,18 @@
+[
+    {
+      "name":         "Test +X magnetic north",
```
Comment (alexshalamov) [inline sensor-tester/build/bundled/src/tests/magnetometer.json]: Test +X points north

File: sensor-tester/build/bundled/src/tests/magnetometer.json [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+      "description":  "Test that sensor reports ~0 μT magnetic field along the X axis when rotate device 360 degrees around each axis before test and then point its +X axis to magnetic north.",
```
Comment (alexshalamov) [inline sensor-tester/build/bundled/src/tests/magnetometer.json]: Test that sensor reports ~0 μT along the Y axis when +X axis points North. Before the test, rotate device 360 degrees around each axis and then point its +X axis to magnetic North.

File: sensor-tester/build/bundled/src/tests/magnetometer.json [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+      "illustration": "src/tests/magnet_+X_north.png",
+      "duration":     5,
+      "expected":     { "x": 0 },
+      "epsilon":      1
+    },
+    {
+      "name":         "Test +Y magnetic north",
```
Comment (alexshalamov) [inline sensor-tester/build/bundled/src/tests/magnetometer.json]: Test +Y points north

File: sensor-tester/build/bundled/src/tests/magnetometer.json [hunk 8]
```diff
@@ -0,0 +1,18 @@
+[
+    {
+      "name":         "Test +X magnetic north",
+      "description":  "Test that sensor reports ~0 μT magnetic field along the X axis when rotate device 360 degrees around each axis before test and then point its +X axis to magnetic north.",
+      "illustration": "src/tests/magnet_+X_north.png",
+      "duration":     5,
+      "expected":     { "x": 0 },
```
Comment (alexshalamov) [inline sensor-tester/build/bundled/src/tests/magnetometer.json]: "expected":     { "y": 0 },

File: sensor-tester/build/bundled/src/tests/magnetometer.json [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+      "epsilon":      1
+    },
+    {
+      "name":         "Test +Y magnetic north",
+      "description":  "Test that sensor reports ~0 μT magnetic field along the Y axis when rotate device 360 degrees around each axis before test and then point its +Y axis to magnetic north.",
```
Comment (alexshalamov) [inline sensor-tester/build/bundled/src/tests/magnetometer.json]: Test that sensor reports ~0 μT along the X axis when +Y axis points North. Before the test, rotate device 360 degrees around each axis and then point its +Y axis to magnetic North.

File: sensor-tester/build/bundled/src/tests/magnetometer.json [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+      "illustration": "src/tests/magnet_+Y_north.png",
+      "duration":     5,
+      "expected":     { "y": 0 },
```
Comment (alexshalamov) [inline sensor-tester/build/bundled/src/tests/magnetometer.json]: "expected":     { "x": 0 },

**Comment by Honry** [comment]
@alexshalamov, thanks! Done.

**Pull request merged**